src/httpserver.cpp

```diff
@@ -215,6 +215,30 @@ static void http_request_cb(struct evhttp_request* req, void* arg)
     {
         WITH_LOCK(g_requests_mutex, g_requests.insert(req));
         g_requests_cv.notify_all();
+
+        auto conn = evhttp_request_get_connection(req);
+
+        // The EV_READ enablement is off-set by the workaround for the libevent
+        // bug further below. This is why it can only be used with libevent
+        // versions that are not affected by the workaround code.
+        if (event_get_version_number() < 0x02010600 && event_get_version_number() >= 0x02020001) {
+            // Enable EV_READ to detect remote disconnection meaning that the close
+            // callback set below is called.
+            bufferevent* bev = evhttp_connection_get_bufferevent(conn);
+            bufferevent_enable(bev, EV_READ);
```

Comment:
I wasn't able to reproduce using this https://github.com/bitcoin/bitcoin/pull/11593#issuecomment-341489496 and since you pointed out that the change wasn't merged into 2.2, I think it's better to err on the side of caution and not enable reading

Reply:
I've updated the PR and removed this